python/cudf/cudf/_libxx/types.pxd

```diff
@@ -8,3 +8,27 @@ ctypedef bool underlying_type_t_order
 ctypedef bool underlying_type_t_null_order
 ctypedef bool underlying_type_t_sorted
 ctypedef int32_t underlying_type_t_interpolation
+
+cdef extern from "cudf/types.hpp" \
+        namespace "cudf" nogil:
+
+    ctypedef enum type_id:
+        EMPTY = 0,
+        INT8,
+        INT16,
+        INT32,
+        INT64,
+        FLOAT32,
+        FLOAT64,
+        BOOL8,
+        TIMESTAMP_DAYS,
+        TIMESTAMP_SECONDS,
+        TIMESTAMP_MILLISECONDS,
+        TIMESTAMP_MICROSECONDS,
+        TIMESTAMP_NAN,
+        STRING,
+        NUM_TYPE_IDS,
+
+    cdef cppclass data_type:
```

Comment:
I think these are already defined in `cpp/types.pxd`. Looks like `type_id` is a C-style "global" enum, but I recall some discussion around enums so there might be a reason for this. @cwharris thoughts?

https://github.com/rapidsai/cudf/blob/3f0859824cae915e246a41be17d4d4789ec72b9f/python/cudf/cudf/_libxx/cpp/types.pxd#L49-L53

Reply:
Also, I think extern C++ declarations should now be placed in the `libcxx/cpp` folder exclusively, per @shwina's guidance.

Reply:
I believe these types are already declared in `_libxx/cpp/types.pxd`. AFAIK, there's no reason we're avoiding enum classes, but some still exist due to legacy reasons (haven't updated that code yet).